language/diem-framework/modules/DiemConsensusConfig.move

```diff
@@ -0,0 +1,78 @@
+address 0x1 {
+
+
+/// Maintains the consensus config for the Diem blockchain. The config is stored in a
+/// DiemConfig, and may be updated by Diem root.
+module DiemConsensusConfig {
+    use 0x1::CoreAddresses;
+    use 0x1::DiemConfig::{Self, DiemConfig};
+    use 0x1::DiemTimestamp;
+    use 0x1::Roles;
+    use 0x1::Vector;
+
+    struct DiemConsensusConfig has copy, drop, store {
+        config: vector<u8>,
+    }
+
+    /// Publishes the DiemConsensusConfig config.
+    public fun initialize(
+        dr_account: &signer,
+    ) {
+        Roles::assert_diem_root(dr_account);
+        DiemConfig::publish_new_config<DiemConsensusConfig>(
+            dr_account,
+            DiemConsensusConfig {config: Vector::empty()},
+        );
+    }
+    spec fun initialize {
+        /// Must abort if the signer does not have the DiemRoot role [[H10]][PERMISSION].
+        include Roles::AbortsIfNotDiemRoot{account: dr_account};
+
+        include DiemConfig::PublishNewConfigAbortsIf<DiemConsensusConfig>;
+        include DiemConfig::PublishNewConfigEnsures<DiemConsensusConfig>{payload: DiemConsensusConfig { config: Vector::empty() }};
+    }
+
+    /// Allows Diem root to update the config.
+    public fun set(dr_account: &signer, config: vector<u8>) {
+        DiemTimestamp::assert_operating();
+
+        Roles::assert_diem_root(dr_account);
+
+        DiemConfig::set<DiemConsensusConfig>(
+            dr_account,
+            DiemConsensusConfig { config }
+        );
+    }
+    spec fun set {
+        /// Must abort if the signer does not have the DiemRoot role [[H10]][PERMISSION].
+        include Roles::AbortsIfNotDiemRoot{account: dr_account};
+
+        include DiemTimestamp::AbortsIfNotOperating;
+        include DiemConfig::SetAbortsIf<DiemConsensusConfig>{account: dr_account};
+        include DiemConfig::SetEnsures<DiemConsensusConfig>{payload: DiemConsensusConfig { config }};
+    }
+
+    // =================================================================
+    // Module Specification
+
+    spec module {} // Switch to module documentation context
+
+    /// # Access Control
+
+    /// Only "set" can modify the DiemConsensusConfig config [[H10]][PERMISSION]
```

Comment:
CC @junkil-park for appropriate updates to DIP2/other access control specs that might need to be updated.

Reply:
In [the permission table](https://github.com/diem/dip/blob/master/dips/dip-2.md#permissions), should we add a new permission entry ("UpdateConsensusConfig | DiemRoot | Y | N") between [H11] and [H12]?

Reply:
there's also InitializeConsensusConfig

Reply:
The permission regarding "ConsensusConfig" will have the label "H12". Could you use "H12" instead of "H10" here? And, also for `InitializeConsensusConfig`?